elasticsearch/docker/build/elasticsearch/bin/docker-entrypoint.sh

```diff
@@ -30,8 +30,8 @@ if [[ "$1" != "eswrapper" ]]; then
         # Without this, user could specify `elasticsearch -E x.y=z` but
         # `bin/elasticsearch -E x.y=z` would not work.
         set -- "elasticsearch" "${@:2}"
-        # Use chroot to switch to UID 1000
-        exec chroot --userspec=1000 / "$@"
+        # Use chroot to switch to non-root UID
+        exec chroot --userspec=${PUID:-${DEFAULT_UID}} / "$@"
```

Comment:
have the chroot's UID set to PUID (defaulting to DEFAULT_UID, which is specified as 1000 in the Dockerfile) rather than just hard-coding 1000